cli/cli.ts

```diff
@@ -4084,212 +4115,263 @@ function testSnippetsAsync(snippets: CodeSnippet[], re?: string, pyStrictSyntaxC
                     throw new Error("codecards must be an JSON array")
                 addSuccess(fn);
             } catch (e) {
-                addFailure(fn, [{
-                    code: 4242,
-                    category: ts.DiagnosticCategory.Error,
-                    messageText: "invalid JSON: " + e.message,
-                    fileName: fn,
-                    start: 1,
-                    line: 1,
-                    length: 1,
-                    column: 1
-                }]);
+                addFailure(
+                    snippet.src,
+                    [createKsDiagnostic(`invalid JSON: ${e.message}`, fn)]
+                );
             }
-            return Promise.resolve();
+            return;
         }
 
-        let isPy = snippet.ext === "py"
-        let inFiles;
-        let extra = snippet.extraFiles || {};
-        if (isPy)
-            inFiles = { "main.ts": "", "main.py": snippet.code, "main.blocks": "", ...extra }
-        else
-            inFiles = { "main.ts": snippet.code, "main.py": "", "main.blocks": "", ...extra }
+        if (++codeSnippetCount % 40 == 0) {
+            // clean up cache every once in a while to speed up / clear up mem
```

Comment:
I fixed this here for now, but we should investigate this further -- it grinds to a complete halt after ~400-500 snippets without this & uses ~2 gb of ram at that point (vs ~400-700mb with this cleanup). Also note the change in service.ts related to reset, seems like that is likely a cause of us crashing at some points during long sessions.